src/analyses/libraryFunctions.ml

```diff
@@ -324,18 +397,44 @@ let svcomp_descs_list: (string * LibraryDesc.t) list = LibraryDsl.[
     ("__VERIFIER_nondet_loff_t", unknown []); (* cannot give it in sv-comp.c without including stdlib or similar *)
   ]
 
+let ncurses_descs_list: (string * LibraryDesc.t) list = LibraryDsl.[
+    ("echo", unknown []);
+    ("noecho", unknown []);
+    ("wattrset", unknown [drop "win" [r_deep; w_deep]; drop "attrs" []]);
+    ("endwin", unknown []);
+    ("wgetch", unknown [drop "win" [r_deep; w_deep]]);
+    ("wmove", unknown [drop "win" [r_deep; w_deep]; drop "y" []; drop "x" []]);
+    ("waddch", unknown [drop "win" [r_deep; w_deep]; drop "ch" []]);
+    ("waddnwstr", unknown [drop "win" [r_deep; w_deep]; drop "wstr" [r]; drop "n" []]);
+    ("wattr_on", unknown [drop "win" [r_deep; w_deep]; drop "attrs" []; drop "opts" []]); (* opts argument currently not used *)
+    ("wrefresh", unknown [drop "win" [r_deep; w_deep]]);
+    ("mvprintw", unknown (drop "win" [r_deep; w_deep] :: drop "y" [] :: drop "x" [] :: drop "fmt" [r] :: VarArgs (drop' [r])));
+    ("initscr", unknown []);
+    ("curs_set", unknown [drop "visibility" []]);
+    ("wtimeout", unknown [drop "win" [r_deep; w_deep]; drop "delay" []]);
+    ("start_color", unknown []);
+    ("use_default_colors", unknown []);
+    ("wclear", unknown [drop "win" [r_deep; w_deep]]);
+    ("can_change_color", unknown []);
+    ("init_color", unknown [drop "color" []; drop "red" []; drop "green" []; drop "blue" []]);
+    ("init_pair", unknown [drop "pair" []; drop "f" [r]; drop "b" [r]]);
+    ("wbkgd", unknown [drop "win" [r_deep; w_deep]; drop "ch" []]);
+  ]
+
 (* TODO: allow selecting which lists to use *)
 let library_descs = Hashtbl.of_list (List.concat [
     c_descs_list;
     posix_descs_list;
     pthread_descs_list;
     gcc_descs_list;
     glibc_desc_list;
+    linux_userspace_descs_list;
     linux_descs_list;
     goblint_descs_list;
     zstd_descs_list;
     math_descs_list;
     svcomp_descs_list;
+    ncurses_descs_list;
```

Comment:
Maybe with this addition it is time to actually make these conditional? With `ncurses` and `zstd` we now have two highly specific libraries in here.

Reply:
The zstd ones are certainly very specific since they are used in the zstd implementation itself. ncurses at least has ones used by actual library users and is a widely used library.

I'll create some activation mechanism for these groups in a follow-up PR. Though it won't be precise until all of the old-style things are converted to new ones in appropriate groups:
https://github.com/goblint/analyzer/blob/bcb4de29e22885570ba82f7ec9cee2e661854b7a/src/analyses/libraryFunctions.ml#L359-L400
https://github.com/goblint/analyzer/blob/bcb4de29e22885570ba82f7ec9cee2e661854b7a/src/analyses/libraryFunctions.ml#L500-L980